tests/test_marshalling.py

```diff
@@ -260,3 +260,27 @@ def test_deserialize_fields_with_dump_only_param(self, unmarshal):
         assert 'years' not in result
 
         assert 'always_invalid' not in unmarshal.errors
+
+    def test_deserialize_wrong_type_root_data(self, unmarshal):
+        data = ''
+        fields_dict = {
```

Comment:
Nitpick: can just be

```suggestion
        fields_dict = {}
```